cmd/server-main.go

```diff
@@ -261,48 +260,36 @@ func isDistributedSetup(eps []*url.URL) bool {
 	return false
 }
 
-// We just exit for invalid endpoints.
-func checkEndpointsSyntax(eps []*url.URL, disks []string) {
+// Check if the endpoints are following expected syntax, i.e
+// valid scheme, valid path across all platforms.
+func checkEndpointsSyntax(eps []*url.URL, disks []string) error {
 	for i, u := range eps {
 		switch u.Scheme {
-		case "", "http", "https":
-			// Scheme is "" for FS and singlenode-XL, hence pass.
+		case "":
+			// "/" is not allowed.
+			if u.Path == "" || u.Path == "/" {
```

Comment:
The reason is when the user runs FS mode => `minio server /`, Empty disk path is not meaningful